mpact_api/mpact/views.py

```diff
@@ -1,3 +1,78 @@
-from django.shortcuts import render
+import os
 
-# Create your views here.
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import ChatData, UserChat, UserData
+from .serializers import UserDataSerializer
+from .services import save_chatdata, save_userchat, save_userdata, set_webhook
+
+
+class Webhook(APIView):
+    """
+    This is only called once for the webhook setup.
+    """
+
+    def get(self, request):
+        webhook = set_webhook()
+        if webhook:
+            return Response("webhook setup ok")
+        else:
+            return Response("webhook setup failed")
+
+
+class ListenMessages(APIView):
+    """
+    Telegram webhook makes a post request.
+    """
+
+    def post(self, request):
+        message = request.data["message"]
```

Comment:
It would be really useful to have examples of what `message`, or `request.data` looks like for different kinds of messages. You could do this in comments, or you could add a few tests, using the Django Test Client to mock incoming webhook calls, and show what this view does.

Reply:
added a sample request.data as a comment